crates/core/tests/helpers/deploy_helpers.rs

```diff
@@ -0,0 +1,476 @@
+use std::collections::HashMap;
+use std::fs;
+use std::fs::File;
+use std::io::Read;
+use std::sync::Arc;
+
+use bytes::BytesMut;
+use dojo_test_utils::sequencer::TestSequencer;
+use dotenv::dotenv;
+use ethers::abi::{Abi, Tokenize};
+use kakarot_rpc_core::client::constants::CHAIN_ID;
+use reth_primitives::{
+    sign_message, Address, Bytes, Transaction, TransactionKind, TransactionSigned, TxEip1559, H256, U256,
+};
+use serde::Deserialize;
+use starknet::accounts::{Account, Call, ConnectedAccount, SingleOwnerAccount};
+use starknet::contract::ContractFactory;
+use starknet::core::chain_id;
+use starknet::core::types::contract::legacy::LegacyContractClass;
+use starknet::core::types::{
+    BlockId, BlockTag, FieldElement, FunctionCall, InvokeTransactionReceipt, MaybePendingTransactionReceipt,
+    TransactionReceipt,
+};
+use starknet::core::utils::{get_contract_address, get_selector_from_name};
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::{JsonRpcClient, Provider};
+use starknet::signers::{LocalWallet, SigningKey};
+use toml;
+use url::Url;
+
+use super::constants::FEE_TOKEN_ADDRESS;
+
+macro_rules! root_project_path {
+    ($relative_path:expr) => {{
+        let crate_root = std::path::Path::new(env!("CARGO_MANIFEST_DIR"));
+        // this is assuming this code will always be ran in the context where the CARGO_MANIFEST_DIR
```

Comment:
I don't understand this sentence

Reply:
i changed the logic use foundry-config's logic and updated the docstring, lmk if it is more clear

https://github.com/foundry-rs/foundry/blob/42fb69461386d58d6f656019aca665c3aed682ae/config/src/utils.rs#L29-L39